packages/autocomplete-plugin-algolia-insights/src/types/AutocompleteInsightsApi.ts

```diff
@@ -6,6 +6,9 @@ export type AutocompleteInsightsApi = ReturnType<
   typeof createSearchInsightsApi
 >;
 
+export type WithArbitraryParams<TParams extends Record<string, unknown>> =
```

Comment:
It's technically an all-purpose utility type, but since it's only being used here for now, it seemed like an unnecessary hassle to move it to `shared`.